### DISCUSSION TITLE
Remove XML tag if only matches specific value

### DISCUSSION THREAD

**Discussion by alwinjmjak** [openrewrite/rewrite#4674] [Q&A] [answered]
I have a xml like below,
```
<arg1>value1</arg1>
<arg1>value2</arg1>
```

I want to remove arg1 tag only if the content matches value1. The current rewrite-xml receipe has [RemoveXmlTag](https://github.com/openrewrite/rewrite/blob/main/rewrite-xml/src/main/java/org/openrewrite/xml/RemoveXmlTag.java) which removes only the specified tags from the xml and does not match the values.

Is there any option to match the content and remove the specified tag from the xml ?

**Comment by timtebeek** [comment, marked as answer]
Hi @alwinjmjak ; Thanks for reaching out! RemoveXmlTag takes in an XPath expression as argument. Have you checked to see if you can give it an expression that would match on tag contents? I haven't tried that myself yet, but it's worth exploring. https://stackoverflow.com/a/2637909

Otherwise adding support for such `//a[text()='Example']` expressions would be the way forward, and any help towards that on a draft PR with tests would be appreciated.

**Reply by alwinjmjak** [reply]
Currently the support to match the content is not available in the XpathMatcher. I'll send out a PR adding the support for it.
